### DISCUSSION TITLE
multi component dynamic providers?

### DISCUSSION THREAD

**Discussion by RonnyPfannschmidt** [reagento/dishka#235] [General]
im the Application im migrating to dishka, we have a plugin system, where between plugins there are shared common component structures/dependencies 

i'd like to be able to define certain dependencies by default for all plugins and wonder how to express it best

a messy idea i had for it would be to have a provider class where component is not set and then creating one instance for each plugin as a component, but i wonder if there is any way to declare them more fluently

```python
class PuginSettings(configObj): ...

class Plugin:

   settings: PluginSettings

class SomePlugin(Plugin, entrypoint="some"): ...

class OtherPlugin(plugin, entrypoint="other")

class PluginBaseProvider(Provider):
    settings = porovide(PluginSettings)
	@provide
	def plugin(self, settings: PluginSettings) -> Plugin
        component = get_component_name_how() # ?
        PluginKlass = get_plugin_class (component)
	   return PluginKlass(settings)


providers = [PlugiBaseProvider(component=entrypoint for entrypoint in find_plugins()]

....
```

**Comment by Tishka17** [comment]
Why not use FromComponent(predifined_one) in plugins?

**Reply by RonnyPfannschmidt** [reply]
The each plugin needs a own defined provider
